TLS.md

```diff
@@ -0,0 +1,104 @@
+# Albatross over TLS
+
+Albatross uses PKI to authenticate both client and server. Requests are signed
+by a certificate authority (CA) that is trusted by the server. CA can delegate
+resources using policies, which happens by creating an intermediate CA.
+Revokation is not implemented, as delegation happens without the server knowing
+about it.
+
+
+This example shows how one can delegate part of the resources to a user. There
+are 4 entities:
+- **server**: runs the albatross TLS endpoint
+- **CA**: root certificate authority
+- **intermediate CA**: user certificate authority
+- **client**: requests initiator
+
+_Note_: there are 4 entities but depending on the security model some can exist
+on the same machine. For example, when **client** and **intermediate CA** can be
+combined, requests are automatically signed using `albatross-client-bistro`.
+
+## Setup
+
+This step-by-step guide shows how files are generated and to which entity they
+belong. Filename is in **bold** when it's created by the current step.
+
+1. Generate the root CA certificate and server keypair
+
+```
+albatross-provision-ca generate ca db
+```
+
+
+| description           | | server         |  CA            | intermediate CA | client |
+| --------------------- |-| -------------- | -------------- | --------------- | ------ |
+| _private key_         | | **server.key** | **ca.key**     |                 |        |
+| _public certificate_  | | **server.pem** | **cacert.pem** |                 |        |
+
+2. **server:** start the endpoint using the server keypair and the root CA certificate
+
+```
+albatross-tls-endpoint cacert.pem server.pem server.key
+```
+
+3. **intermediate CA:** we want to delegate part of the resources to a given
+user. The user generates a signing request to allow a memory of 1024MB to run
+16 unikernels on two CPUs.
+
+```
+albatross-provision-request add_policy user 16 --mem 1024 --cpu 0
```

Comment:
```suggestion
albatross-provision-request add_policy user 16 --mem 1024 --cpu 0 --cpu 1
```

In the text, there's `two CPUs` -- that should be reflected in the command.

Also I'm wondering whether we need a slight excursion what these `cpu` arguments are meaning / what they're used for.

Reply:
Good catch, thanks. Indeed I was confused by the `--cpu <n>` option at first I thought `<n>` was the total number of CPUs allowed, instead of the core ID. I'll make the doc more precise on that matter.

Reply:
I have added a commit to make the doc consistent and replaced `two CPUs` by `CPU IDs 0 and 1`. It still not very clear what that means as I'm not familiar with the option, so don't hesitate to add something if you want !